build/notarize.js

```diff
@@ -23,20 +23,39 @@ exports.default = async function notarizing(context) {
   const appName = context.packager.appInfo.productFilename;
   const { SIGNING_APPLE_ID, SIGNING_APP_PASSWORD, SIGNING_TEAM_ID } = process.env;
 
-  if (isEmpty(SIGNING_APPLE_ID) || isEmpty(SIGNING_APP_PASSWORD)) {
-    log('SIGNING_APPLE_ID or SIGNING_APP_PASSWORD not set.\nTerminating noratization.');
+  if (isEmpty(SIGNING_APPLE_ID)) {
+    log('SIGNING_APPLE_ID not set.\nTerminating notarization.');
+    return;
+  }
+
+  if (isEmpty(SIGNING_APP_PASSWORD)) {
+    log('SIGNING_APP_PASSWORD not set.\nTerminating notarization.');
     return;
   }
 
-  const options = {
-    appBundleId: 'com.loki-project.messenger-desktop',
-    appPath: `${appOutDir}/${appName}.app`,
-    appleId: SIGNING_APPLE_ID,
-    appleIdPassword: SIGNING_APP_PASSWORD,
-  };
-  if (!isEmpty(SIGNING_TEAM_ID)) {
-    options.ascProvider = SIGNING_TEAM_ID;
-    options.teamId = SIGNING_TEAM_ID;
+  if (isEmpty(SIGNING_TEAM_ID)) {
+    log(' SIGNING_TEAM_ID not set.\nTerminating notarization.');
+    return;
   }
-  return notarize(options);
+
+  const appPath = `${appOutDir}/${appName}.app`;
+  const zipPath = `${appOutDir}/${appName}.zip`;
+  const appleId = SIGNING_APPLE_ID;
+  const appleIdPassword = SIGNING_APP_PASSWORD;
+  const teamId = SIGNING_TEAM_ID;
+
+  console.log(
```

Comment:
Can we add a NOTE comment with a link to this solution that you found please? Gives context if we comeback to this in future 🙏

Reply:
Also maybe write that Apple's notarization service expects an unflattened zip ofthe .app file to be submitted?

Reply:
added official info link